v1/plugin/session.go

```diff
@@ -0,0 +1,117 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"net/http/pprof"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+var (
+	listenPort = "0"
+	LogLevel   = uint8(2)
+	pprofPort  = "0"
+)
+
+// Arguments passed to startup of Plugin
+type Arg struct {
+	// Plugin log level, see logrus.Loglevel
+	LogLevel uint8
+	// Ping timeout duration
+	PingTimeoutDuration time.Duration
+
+	// The listen port
+	ListenPort string
+
+	// enable pprof
+	Pprof bool
+}
+
+// getArgs returns plugin args or default ones
+func getArgs() error {
+	pluginArg := &Arg{}
+	if os.Args[1] == "" {
+		return nil
```

Comment:
Some medium tests are not passing anything.  All those arguments are optional, I would understand if we start a plugin without those, especially for testing purposes.